crates/kitsune_p2p/dht/src/lib.rs

```diff
@@ -0,0 +1,180 @@
+//! implies a collection of uniform grids layered on top of the quantum grid, each one twice
+//! as coarse as the one below it.
+//!
+//! Thus, we can refer to the successive levels of coarseness of quantization by a single
+//! whole number, referred to as the "power", since it takes the form of `2^p`.
+//! A "power" of 0 refers to the quantum grid itself. A power of 1 is twice as coarse, and so on.
+//!
+//! Arcs are specified as a set of contiguous Segments all of the same power.
+//!
+//! ## Quantized Gossip
+//!
+//! The reason for this detailed quantization of spacetime is to implement historical gossip between
+//! agents efficiently. Each Agent has its own Arc which represents which Ops it is responsible
+//! for storing and gossiping to other agents. When two agents gossip, they have to quickly and
+//! efficiently determine which Ops they already have in common, and which they need to send
+//! to each other.
+//!
+//! This as accomplished by agents exchanging information about the Regions of spacetime that they
+//! hold in common. This needs to be done with a minimum of coordination, which is why we use a
+//! uniform quantization of spacetime -- if there are only so many ways to split up spacetime into
+//! Regions, there is a greater chance that agents will have information on the exact same regions.
+//!
+//! Concretely, when agents gossip, they exchange the "fingerprint" of each region they have in common
+//! with their counterpart, and for any regions whose fingerprints mismatch, each agent sends all
+//! the ops in that region to their counterpart. For the regions which match, it is understood that
+//! both parties hold the same data in those regions and need no further gossip on those regions.
+//! Therefore, it is important to split spacetime into regions in such a way that when mismatches in
+//! region fingerprints occur, they are for small amounts of data; or if they are for large amounts
+//! of data, then the mismatches should be rare. At the same time, we don't want to split into too
+//! many regions, because that would lead to a larger overhead with minimal gains.
+//!
+//! ### Quantizing space
+//!
+//! Each agent selects a power level for its arc based on its observation of the network conditions.
+//! Ideally, each agent sees a similar view of the network and chooses the same power level as the
+//! other nearby agents. When the power levels match, the two agents are tracking space at the same
+//! level of granularity and can more easily coordinate. Inasmuch as the power levels differ,
+//! one of the agents will have to do some extra computation to reconcile the difference.
+//!
+//! Without going into too much detail, the power level is mainly a function of how many other
+//! Agents are in the vicinity. Agents start with a high power level, covering large sections of
+//! the DHT with a coarsely subdivided arc, and as more agents join, they reduce their power level,
```

Comment:
If you reduce the power level, that results in less cells, which in my understanding equals a coarser granularity, whereas an increased power level leads to more cells and therefore finer resolution. 🤔

Reply:
It is the opposite of what you wrote. The length of the edges of each grid cell is determined by `2^power`, so a higher power means a coarser grid. A power of 0 corresponds to the quantum level, the finest resolution possible. Increased power results in fewer cells.